README.md

```diff
@@ -170,133 +170,28 @@ example by changing the class declared in main.cpp.
-  cards or eMMC chips offer a full FTL layer on top of NAND flash. This makes the
-  storage well-suited for systems that require a about 1GB of memory.
-  Additionally, SD cards are a popular form of portable storage. They are useful
-  if you want to store data that you can access from a PC.
-  
-  ``` cpp
-  SDBlockDevice bd(
-          MBED_CONF_SD_SPI_MOSI,
-          MBED_CONF_SD_SPI_MISO,
-          MBED_CONF_SD_SPI_CLK,
-          MBED_CONF_SD_SPI_CS);
-  ```
-
-- [**HeapBlockDevice**](https://os.mbed.com/docs/v5.6/reference/heapblockdevice.html) -
-  Block device that simulates storage in RAM using the heap. Do not use the heap
-  block device for storing data persistently because a power loss causes
-  complete loss of data. Instead, use it fortesting applications when a storage
-  device is not available.
-  
-  ``` cpp
-  HeapBlockDevice bd(1024*512, 512);
-  ```
-
-Additionally, Mbed OS contains several utility block devices to give you better
-control over the allocation of storage.
-
-- [**SlicingBlockDevice**](https://os.mbed.com/docs/latest/reference/slicingblockdevice.html) -
-  With the slicing block device, you can partition storage into smaller block
-  devices that you can use independently.
-
-- [**ChainingBlockDevice**](https://os.mbed.com/docs/latest/reference/chainingblockdevice.html) -
-  With the chaining block device, you can chain multiple block devices together
-  and extend the usable amount of storage.
-
-- [**MBRBlockDevice**](https://os.mbed.com/docs/latest/reference/mbrblockdevice.html) -
-  Mbed OS comes with support for storing partitions on disk with a Master Boot
-  Record (MBR). The MBRBlockDevice provides this functionality and supports
-  creating partitions at runtime or using preformatted partitions configured
-  separately from outside the application.
-
-- **ReadOnlyBlockDevice** - With the read-only block device, you can wrap a
-  block device in a read-only layer, ensuring that user of the block device does
-  not modify the storage.
-
-- **ProfilingBlockDevice** - With the profiling block device, you can profile
-  the quantity of erase, program and read operations that are incurred on a
-  block device.
-
-- **ObservingBlockDevice** - The observing block device grants the user the
-  ability to register a callback on block device operations. You can use this to
-  inspect the state of the block device, log different metrics or perform some
-  other operation.
-
-- **ExhaustibleBlockDevice** - Useful for evaluating how file systems respond to
-  wear, the exhaustible block device simulates wear on another form of storage.
-  You can configure it to expire blocks as necessary.
-
-## Tested configurations
+Alternatively, you may also choose to reimplement the function `BlockDevice::get_default_instance()`, but this is more involved and isn't recommended unless you have a good understanding of block devices and the overall BlockDevice API.
```

Comment:
```suggestion
Alternatively, you may use the system's default block device `BlockDevice *bd = BlockDevice::get_default_instance()` but this will require more code changes to the example.
```

Reply:
The original text more accurately represents the last option IMO. Here is the explanation taken from BlockDevice.h. 
``` cpp
     * An application can override all target settings by implementing
     * BlockDevice::get_default_instance() - the default
     * definition is weak, and calls get_target_default_instance().
 ```

Reply:
The example does not currently use BlockDevice::get_default_instance. So instead of instantiated a block device, it could get the default block device. It could also reimplement that function as you suggest, which is yet another step.

Reply:
I see, it probably is better to mention this at the very start before discussing ways to add non-default devices.

Reply:
For the sake of simplicity and without going into too much detail, I propose the above. I think this gives a good starting point for someone who would like to change the default implementation which they will know is possible since this is a weak function.